### ISSUE TITLE
Undefined values, as no longer set up on componentWillMount

### ISSUE THREAD

**Issue by alistairjcbrown** [JedWatson/react-select#303]
Issue introduced in the latest version `v0.5.3` (not present in `v0.5.2`)

```
Uncaught TypeError: Cannot read property 'forEach' of undefined
```
Which references this line: https://github.com/JedWatson/react-select/blob/23f0436094432986392aadbb1482cd12ef36a627/lib/Select.js#L720

Looks like this is related to the change in PR #280
https://github.com/JedWatson/react-select/pull/280/files#diff-04191649796841200b33375bdff7c78fL133

This PR removes the state setup code which sets up `values` - there is a comment in the removed code which references `Fixes issue #201` - #201 also details that `this.state.values` is not set up, causing an error.

Manually adding the removed code back in solves the issue on my build, with the latest version `v0.5.3`.

**Comment by alistairjcbrown** [comment]
@JedWatson I think too much was code removed in that last version which included changes to `autoloadAsyncOptions` - I've added a PR #304 which should fix this issue.

**Comment by JedWatson** [comment]
Fixed by #304
